### PR TITLE
Silenced -Warray-bounds warnings.

### PR DESCRIPTION
**Pull request by rohit-n**

### CODE REVIEW

File: Source/Core/Core/DSP/DSPAssembler.cpp [hunk 1]
```diff
@@ -706,6 +706,13 @@ bool DSPAssembler::VerifyParams(const opc_t *opc, param_t *par, int count, bool
 // Merge opcode with params.
 void DSPAssembler::BuildCode(const opc_t *opc, param_t *par, u32 par_count, u16 *outbuf)
 {
+	if (par_count > sizeof(opc->params) / sizeof(opc->params[0]))
```
Comment (lioncash) [inline Source/Core/Core/DSP/DSPAssembler.cpp]: `>=`

**Comment by lioncash** [comment]
BTW I also sent an invite to be part of the trusted devs group so no one needs to kick the buildbot for your PRs. Sorta overdue, really.

**Comment by rohit-n** [comment]
Thanks.
@dolphin-emu-bot rebuild

**Comment by comex** [comment]
Do you have a link to the warning text?  (I couldn't find -Warray-count on Google; do you mean array-bounds or something?)
Why is GCC complaining about this?  Is it actually possible for it to be out of bounds?  If not then I'm not impressed it wants a pointless check; and such a check should be at most an _assert_.

**Comment by rohit-n** [comment]
You're right, it's -Warray-bounds. The text is "array subscript is above array bounds", and it occurs throughout the for loop. I think only GCC 5 is complaining, since I've only seen it very recently.

**Comment by comex** [comment]
I think the real problem here is that GetParams goes up to 10, but DSPOPCTemplate::params is of count 8.

**Comment by rohit-n** [comment]
Sounds like increasing the size of params would be best, then?

**Comment by rohit-n** [comment]
I'm not getting this warning any more, so I'll close this.

**Pull request closed without merging**